var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -16,12 +16,15 @@ class PyCupy(PythonPackage):
     homepage = "https://cupy.dev/"
     pypi = "cupy/cupy-8.0.0.tar.gz"
 
+    version("11.2.0", sha256="c33361f117a347a63f6996ea97446d17f1c038f1a1f533e502464235076923e2")
     version("8.0.0", sha256="d1dcba5070dfa754445d010cdc952ff6b646d5f9bdcd7a63e8246e2472c3ddb8")
 
-    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"))
```

Comment:
You should make the new dependencies conditional on version, so:
```python
depends_on("python@3.7:", when="@11:", type=("build", "run"))
depends_on("python@3.6:", type=("build", "run"))
```
or something like that. Same for other changed deps.

Reply:
My feeling is to ditch the 8.0 tagged version to be honest, because it's ancient.

Reply:
In that case you should deprecate it, not remove it.